src/lightning/fabric/strategies/xla_fsdp.py

```diff
@@ -0,0 +1,428 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+from contextlib import contextmanager
+from pathlib import Path
+from typing import Any, Callable, Dict, Generator, List, Literal, Optional, Tuple, Union
+
+import torch
+from torch import Tensor
+from torch.nn import Module
+from torch.optim import Optimizer
+from torch.utils.data import DataLoader
+
+from lightning.fabric.accelerators import Accelerator
+from lightning.fabric.accelerators.xla import _XLA_AVAILABLE
+from lightning.fabric.plugins.io.checkpoint_io import CheckpointIO
+from lightning.fabric.plugins.precision import Precision
+from lightning.fabric.strategies import XLAStrategy
+from lightning.fabric.strategies.strategy import _BackwardSyncControl, _validate_keys_for_strict_loading
+from lightning.fabric.utilities.imports import _TORCH_GREATER_EQUAL_2_0
+from lightning.fabric.utilities.rank_zero import rank_zero_only, rank_zero_warn
+from lightning.fabric.utilities.types import _PATH, Optimizable
+
+if _XLA_AVAILABLE:
+    import torch_xla.core.xla_model as xm
+    from torch_xla.distributed.fsdp import XlaFullyShardedDataParallel as XLAFSDP
+    from torch_xla.distributed.parallel_loader import MpDeviceLoader
+else:
+    XLAFSDP = None
+    MpDeviceLoader = None
+    xm = None
+
+
+class XLAFSDPStrategy(XLAStrategy):
+    """Strategy for training multiple TPU devices using the
+    :func:`torch_xla.distributed.xla_fully_sharded_data_parallel.XlaFullyShardedDataParallel` method.
+
+    For more information check out https://github.com/pytorch/xla/blob/master/docs/fsdp.md
+    """
+
+    def __init__(
+        self,
+        accelerator: Optional[Accelerator] = None,
+        parallel_devices: Optional[List[torch.device]] = None,
+        checkpoint_io: Optional[CheckpointIO] = None,
+        precision: Optional[Precision] = None,
+        auto_wrap_policy: Optional[Callable] = None,
+        auto_wrapper_callable: Optional[Callable] = None,
+        state_dict_type: Literal["full", "sharded"] = "sharded",
+        sync_module_states: bool = True,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(
+            accelerator=accelerator,
+            parallel_devices=parallel_devices,
+            checkpoint_io=checkpoint_io,
+            precision=precision,
+        )
+        self._backward_sync_control = _XLAFSDPBackwardSyncControl()
+
+        if auto_wrap_policy:
+            kwargs["auto_wrap_policy"] = auto_wrap_policy
+        if auto_wrapper_callable:
+            kwargs["auto_wrapper_callable"] = auto_wrapper_callable
```

Comment:
Why not get these directly from kwargs in the signature?

Reply:
iiuc, the suggestion here is to remove `auto_wrap_policy` and `auto_wrap_callable` as function arguments and rather collect them if they are in kwargs?

Reply:
Thats right